dozer-ingestion/mysql/src/binlog.rs

```diff
@@ -434,14 +433,32 @@ impl BinlogIngestor<'_, '_, '_> {
                         }
 
                         let table = table_cache.get_table_details(table_index).unwrap();
-                        self.handle_rows_event(&rows_event, &table, tme).await?;
+
+                        self.handle_rows_event(
+                            &rows_event,
+                            &table,
+                            tme,
+                            self.ingestion_start_position.clone(),
+                            &mut transaction_pos,
+                        )
+                        .await?;
+
+                        if let Some(start_binlog_position) = &self.ingestion_start_position {
+                            if start_binlog_position < &transaction_pos {
+                                self.ingestion_start_position = None;
+                            }
+                        }
                     }
                 }
 
                 event_type => {
                     trace!("other binlog event {event_type:?}");
                 }
             }
+
+            if binlog_event.header().log_pos() > 0 {
+                prev_position = binlog_event.header().log_pos().into();
+            }
```

Comment:
can this become `prev_position = self.next_position.position`?

Reply:
To make it clearer it can be move right before this line https://github.com/getdozer/dozer/blob/44d5130cd01b836de597b19fbd8d8c4c041eecab/dozer-ingestion/mysql/src/binlog.rs#L187

Reply:
Completely removed this `prev_position = self.next_position.position` part. 

I have done it to store `log_pos` of transaction start (btw, `log_pos` is 0 for first event in log). So now I just calculate  it using 
```
transaction_pos.position = (binlog_event.header().log_pos()
      - binlog_event.header().event_size())
      as u64;
```

Example of first two messages
```
INFO Ingestion starting at BinlogPosition { filename: [98, 105, 110, 108, 111, 103, 46, 48, 48, 48, 50, 53, 56], position: 9816006, seq_no: 12 }
 INFO Header: BinlogEventHeader {
    timestamp: 1704611411,
    event_type: FORMAT_DESCRIPTION_EVENT,
    server_id: 1,
    event_size: 121,
    log_pos: 0,
    flags: EventFlags(0x0),
}
 INFO Header: BinlogEventHeader {
    timestamp: 1705407102,
    event_type: QUERY_EVENT,
    server_id: 1,
    event_size: 95,
    log_pos: 9816101,
    flags: EventFlags(LOG_EVENT_SUPPRESS_USE_F),
}
```

So now, when parsing query event, we calculate start log_pos by `log_pos` - `event_size`